src/game/gameflow.c

```diff
@@ -1173,21 +1174,32 @@ GameFlow_InterpretSequence(int32_t level_num, GAMEFLOW_LEVEL_TYPE level_type)
                 while (Output_FadeIsAnimating()) {
                     Output_InitialisePolyList();
                     Output_CopyPictureToScreen();
+                    Input_Update();
                     Output_DumpScreen();
+                    if (g_Input.any) {
+                        break;
+                    }
                 }
 
-                Output_InitialisePolyList();
-                Output_CopyPictureToScreen();
-                Output_DumpScreen();
-                Shell_Wait(data->display_time);
+                if (!g_InputDB.any) {
+                    Output_InitialisePolyList();
+                    Output_CopyPictureToScreen();
+                    Output_DumpScreen();
+                    Shell_Wait(data->display_time);
+                }
 
                 // fade out
                 Output_FadeToBlack(true);
                 while (Output_FadeIsAnimating()) {
```

Comment:
Gonna be annoying and say maybe make all three input checks consistent? (unless there's a reason they are structured differently).

`while (Output_FadeIsAnimating() && !g_InputDB.any)`?

Reply:
Simplified